torchaudio/models/wav2vec2/components.py

```diff
@@ -426,11 +453,28 @@ def forward(
             # extend the mask to attention shape and set weight
             mask = -10000.0 * mask[:, None, None, :].to(dtype=features.dtype)
             mask = mask.expand(batch_size, 1, max_len, max_len)
+        return x, mask
 
+    def forward(
+            self,
+            features: Tensor,
+            lengths: Optional[Tensor] = None,
+    ) -> Tensor:
+        x, mask = self._preprocess(features, lengths)
         x = self.transformer(x, attention_mask=mask)
         x = self.readout(x)
         return x
 
+    def extract_feature(
```

Comment:
I share the sentiment of confusion, however the following is my thoughts

1. This method is the one attached on `Encoder` class, which will be used by `Wave2Vec2Model` class. The expected usage is via `Wav2Vec2Model.extract_features`.
2. This class is already called `Encoder`, so renaming the other module to `Encoder` might be more confusing.
3. Since the original `fairseq` call the other module `FeatureExtractor`, I think it is better to keep it that way.
4. The word `encode[r]` is overloaded.
5. From the viewpoint of encoder, the fact that the internal implementation uses transformer is better abstracted away.

> it's plural

Yes, let me fix that one.

Reply:
Got it. From users' viewpoint, they will only touch ``Wav2Vec2Model.extract_features`` and ``Wav2Vec2Model.forward`` methods. So this should be fine.